src/CSModule.sol

```diff
@@ -721,7 +729,56 @@ contract CSModule is
             incrementNonceIfUpdated: true
         });
         // Direct call of `enqueueNodeOperatorKeys` if depositable is not changed
-        depositQueue.enqueueNodeOperatorKeys(_nodeOperators, nodeOperatorId);
+        _enqueueNodeOperatorKeys(nodeOperatorId);
+    }
+
+    function _enqueueNodeOperatorKeys(uint256 nodeOperatorId) internal {
+        uint256 curveId = accounting.getBondCurveId(nodeOperatorId);
+        (
+            uint32 queuePriority,
+            uint32 priorityQueueMaxKeys
+        ) = PARAMETERS_REGISTRY.getPriorityQueueConfig(curveId);
+
+        NodeOperator storage no = _nodeOperators[nodeOperatorId];
+        uint32 enqueuedSoFar = no.totalDepositedKeys + no.enqueuedCount;
+
+        if (priorityQueueMaxKeys > enqueuedSoFar) {
+            uint32 leftForPriorityQueue = priorityQueueMaxKeys - enqueuedSoFar;
+            _enqueueNodeOperatorKeys(
+                nodeOperatorId,
+                queuePriority,
+                leftForPriorityQueue
+            );
+        }
+
+        _enqueueNodeOperatorKeys(
+            nodeOperatorId,
+            LOWEST_PRIORITY,
+            type(uint32).max
+        );
+    }
+
+    function _enqueueNodeOperatorKeys(
```

Comment:
Why not keep it in lib?

Reply:
Because it hardly depends on the queue internal implementation. We only use `enqueue` method, which is public facing API, and I found it easier to have in the module instead.

Reply:
Bytecode economy?

Reply:
18 bytes.